src/App.jsx

```diff
@@ -14,45 +17,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ABC = 'abc';
+const SORT_FIELD_LENGTH = 'length';
+let REVERSE = false;
+let ABC = false;
+let LENGTH = false;
+
+export const App = () => {
+  let visibleGoods = [...goodsFromServer];
+  const [count, setCount] = useState(0);
+  const [sortField, setSortField] = useState('');
+  const hasResetButton = REVERSE || sortField;
+
+  const handleSort = field => () => setSortField(field);
+
+  if (sortField === SORT_FIELD_ABC) {
+    LENGTH = false;
+    ABC = true;
+  } else if (sortField === SORT_FIELD_LENGTH) {
+    LENGTH = true;
+    ABC = false;
+  } else if (!sortField && !count) {
+    LENGTH = false;
+    ABC = false;
+    REVERSE = false;
+  }
```

Comment:
don't need to create this redundant variable outside the component 
you have already sortField, so you need to use it